k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -803,7 +803,7 @@ Upload any local messages that are marked as PENDING_UPLOAD (Drafts, Sent, Trash
 
                 if (Expunge.EXPUNGE_ON_POLL == account.getExpungePolicy()) {
                     Timber.d("SYNC: Expunging folder %s:%s", account.getDescription(), folder);
-                    remoteFolder.expunge();
+                    remoteFolder.expunge(localFolder.getAllMessageUids(true));
```

Comment:
Eventually I want to get rid of 'expunge on sync'. For now I think preserving the current behavior by calling `expunge()` rather than `expungeUids()` is fine.